Propagate worker proxy settings when building docker images

boazy opened the pull request:
Solves issue #131

Concourse supports passing the http_proxy family of environment variables from the workers to the containers running on them: concourse/concourse#319

These environment variables do not propagate, however, when building a docker image with docker-image-resource which leaves a pretty large blind spot. This PR fixes that by adding all present proxy setting environment variables as build arguments.

You may see this warning during docker image build if you have proxy environment variable enabled:
`[Warning] One or more build-args [http_proxy https_proxy] were not consumed`

This is because Docker fails to detect that these arguments are being used (even when they're used). It's thankfully harmless.

pivotal-issuemaster commented:
@boazy Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=141)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=141) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-issuemaster commented:
@boazy Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=141)!

jtarchie commented on assets/out:
```diff
@@ -112,6 +112,14 @@ elif [ -n "$build" ]; then
+  done
+
   if [ -n "$build_args" ]; then
     expanded_build_args=$(echo $build_args|jq -r 'with_entries(.key |= "--build-arg " + . )|with_entries(.key = .key + "=" +.value)|keys|join(" ")')
```
This line would squash your appended values. This wouldn't break any existing pipelines, so it is backwards compatible, but wouldn't you want the proxy values always?

boazy replied:
Oh, didn't notice that! Thanks.
I've fixed that now. I hope.

boazy commented:
@jtarchie i fixed the pull request, can you review again?

vito commented:
Ugh, I have no idea what's up with GitHub's conflict resolution thing (that's what made the last commit). It never works. It makes a commit but then the conflict is *still* not resolved.

...In any case, this is now pretty broken via conflicts with https://github.com/concourse/docker-image-resource/pull/159 and is still in need of tests. I tried to resolve the conflicts and switch it to the new buid_arg flow, but that didn't work. Do you have time to update this, @boazy? Thanks!

boazy commented:
@vito Thanks for resolving the conflict and sorry for neglecting the unit tests.
I didn't know how to merge back everything nicely into my branch, so I've just rebased everything again against the current master (with your fixes). I hope that's okay.
I've also added unit tests now.

vito commented:
Sweet, thanks for following up! Tests look good, I'll go ahead and merge.
